python/ray/data/dataset.py

```diff
@@ -508,6 +505,7 @@ def transform(
                 transform,
                 compute,
                 ray_remote_args,
+                target_block_size=batch_size,
```

Comment:
We may want to take the `ctx.target_max_block_size` into account here.

Reply:
I think it will be important to cap byte size of target block here. Take image read as an example, if user read images (256\*256) with default batch size (4096), it could be more than 800MB per batch in minimum (256\*256\*3\*4096).

Reply:
Yeah, but I don't know if that `ctx.target_max_block_size` should be a strict cap in this case, since a user might have e.g. some very beefy nodes with lots of CPU memory, and they may want to be churning through multi-gigabyte batches, with GPU-based inference as an example.

Reply:
shall we add a TODO here? we know we want to cap the byte size, but the threshold is still underdertermining.